src/wp-includes/class-wp-scripts.php

```diff
@@ -418,15 +464,17 @@ public function do_item( $handle, $group = false ) {
 	 * Adds extra code to a registered script.
 	 *
 	 * @since 4.5.0
+	 * @since 6.3.0 Addition of $standalone boolean parameter.
 	 *
-	 * @param string $handle   Name of the script to add the inline script to.
-	 *                         Must be lowercase.
-	 * @param string $data     String containing the JavaScript to be added.
-	 * @param string $position Optional. Whether to add the inline script
-	 *                         before the handle or after. Default 'after'.
+	 * @param string $handle     Name of the script to add the inline script to.
+	 *                           Must be lowercase.
+	 * @param string $data       String containing the JavaScript to be added.
+	 * @param string $position   Optional. Whether to add the inline script
+	 *                           before the handle or after. Default 'after'.
+	 * @param bool   $standalone Optional. Inline script opted to be standalone or not. Default false.
```

Comment:
This optional parameter doesn't really help developers reason about "why" they would want to use this parameter, so I think we should attempt to make this more descriptive.

Related, as we're iterating on the way both `async` scripts and inline scripts are handled in this PR, we need to be open to asking whether the original concerns this addition meant to address are still present and worthy of adding additional complexity to this API in order to address.

My current understanding is that at present, this has two effects:
1. An inline script added after a deferred script will not execute until the deferred script is executed to ensure execution order remains the same.
2. If an inline script is added to a script handle in the before position, that script handle and all it's dependencies will be changed into blocking scripts in `get_eligible_loading_strategy()`

The addition of this parameter allows scripts to avoid these back-compatibility constraints by declaring that the script doesn't depend on the execution order of the handle it is registered to or any of that handle's dependencies. If we determine that we've handled these use cases elsewhere, or if we're not overly concerned with the need for an opt-in way to avoid back-compat constraints, then I think we could remove this to reduce complexity overall.